internal/rules/no_unnecessary_type_assertion/no_unnecessary_type_assertion_test.go

```diff
@@ -1008,8 +1008,7 @@ const foo = (  3 + 5
       `,
 			Output: []string{`
 const foo = (  3 + 5
-  ) /*as*/  //as
-  ;
```

Comment:
hmm this is technically a regression

Reply:
If we consider:
```
const foo  =
    initialiser /* comment 1 */
as  /* comment 2 */
typeAssertion  /* comment 3 */
```

We can probably take the following approach:
1. create two "fixes, the first to remove the `as` keyword, the second to remove `typeAssertion`
2. we merge the two fixes together if the text between them is only whitespace
3. we move the start of the two fixes earlier if the text before it is white space